spotlight-client/src/VizPrisonStayLengths/VizPrisonStayLengths.test.tsx

```diff
@@ -0,0 +1,153 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2021 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import {
+  fireEvent,
+  screen,
+  waitForElementToBeRemoved,
+  within,
+} from "@testing-library/react";
+import { runInAction, when } from "mobx";
+import React from "react";
+import DemographicsByCategoryMetric from "../contentModels/DemographicsByCategoryMetric";
+import DataStore from "../DataStore";
+import { reactImmediately, renderWithStore } from "../testUtils";
+import { colors } from "../UiLibrary";
+import VizPrisonStayLengths from "./VizPrisonStayLengths";
+
+jest.mock("../MeasureWidth/MeasureWidth");
+
+let metric: DemographicsByCategoryMetric;
+
+beforeEach(() => {
+  runInAction(() => {
+    DataStore.tenantStore.currentTenantId = "US_ND";
+  });
+  reactImmediately(() => {
+    const metricToTest = DataStore.tenant?.metrics.get(
+      "PrisonStayLengthAggregate"
+    );
+    // it will be
+    if (metricToTest instanceof DemographicsByCategoryMetric) {
+      metric = metricToTest;
+    }
+  });
+});
+
+afterEach(() => {
+  runInAction(() => {
+    // reset data store
+    metric.demographicView = "total";
+    DataStore.tenantStore.currentTenantId = undefined;
+  });
+});
+
+test("loading", () => {
```

Comment:
hmm yeah I haven't been doing that due in part to laziness (error handling is tested in the Metric model but not in the UI) but also because the error handling is still kind of a placeholder ... there is a ticket #327 to flesh it out, I made a note there about the existing test coverage because it may depend on what the real designs are